packages/core/src/Stores/Modules/Cashier/cashier-store.js

```diff
@@ -210,14 +210,15 @@ export default class CashierStore extends BaseStore {
                 this.root_store.client.is_logged_in,
                 this.root_store.client.currency,
             ],
-            () => {
+            async () => {
+                await BinarySocket.wait('get_settings');
```

Comment:
Can you add a comment here why it's needed? Because it seems like this is just a hack 😄

Reply:
Ok, I added a comment in the code. The actual problem is that it has some 1-4 seconds delay before most of the values are populated in the client-store, so on some stages of our system it's impossible to determine what account or f.e. residence a user has. I came across this problem previously when I worked on route changes as well. Maybe we can think of a way to keep some settings always in the system (like `client residence`) and get some settings from our side directly (like `account` details).